src/DateTime.swift

```diff
@@ -145,16 +145,16 @@ public struct DateTime: Comparable
 
     :returns: A new instance of DateTime.
     */
-    public init(date: NSDate, timezone: NSTimeZone = NSTimeZone.defaultTimeZone())
+    public init(date: Date, timezone: TimeZone = TimeZone.current)
     {
         self.internalDate = date
         self.timezone = timezone
 
-        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+        var calendar = Calendar(identifier: Calendar.Identifier.gregorian)
         calendar.timeZone = timezone
 
-        let calendarUnits = NSCalendarUnit(rawValue: UInt.max)
-        self.calendarComponents = calendar.components(calendarUnits, fromDate: self.internalDate)
+        let calendarUnits = Set<Calendar.Component>([.day, .hour, .year, .minute, .second, .weekday, .weekdayOrdinal, .weekOfMonth, .weekOfYear])
```

Comment:
@Haud Question. With Swift 3, you can no longer do `NSCalendarUnit(rawValue: UInt.max)` to automatically get all the cases -- how did you want to handle this?